planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp

```diff
@@ -215,49 +50,52 @@ bool StopLineModule::modifyPathVelocity(
 
   const LineString2d stop_line = planning_utils::extendLine(
     stop_line_[0], stop_line_[1], planner_data_->stop_line_extend_length);
-  const auto & current_position = planner_data_->current_pose.pose.position;
-  const PointWithSearchRangeIndex src_point_with_search_range_index =
-    planning_utils::findFirstNearSearchRangeIndex(path->points, current_position);
-  SearchRangeIndex dst_search_range =
-    planning_utils::getPathIndexRangeIncludeLaneId(*path, lane_id_);
-
-  // extend following and previous search range to avoid no collision
-  if (dst_search_range.max_idx < path->points.size() - 1) dst_search_range.max_idx++;
-  if (dst_search_range.min_idx > 0) dst_search_range.min_idx--;
 
-  // Find collision
-  const auto collision = findCollision(*path, stop_line, dst_search_range);
+  // Calculate stop pose and insert index
+  const auto stop_point = arc_lane_utils::createTargetPoint(
+    *path, stop_line, lane_id_, planner_param_.stop_margin,
+    planner_data_->vehicle_info_.max_longitudinal_offset_m);
 
   // If no collision found, do nothing
-  if (!collision) {
+  if (!stop_point) {
     RCLCPP_DEBUG_THROTTLE(logger_, *clock_, 5000 /* ms */, "is no collision");
     return true;
   }
-  const double center_line_z = (stop_line_[0].z() + stop_line_[1].z()) / 2.0;
-  const auto stop_line_position = planning_utils::toRosPoint(collision->point, center_line_z);
-
-  // Find offset segment
-  const auto offset_segment = findOffsetSegment(*path, *collision);
 
-  // Calculate stop pose and insert index
-  const auto stop_pose_with_index = calcStopPose(*path, offset_segment);
+  const auto stop_point_idx = stop_point->first;
+  auto stop_pose = stop_point->second;
+  stop_pose.position.z = (stop_line_[0].z() + stop_line_[1].z()) / 2.0;
 
-  const PointWithSearchRangeIndex dst_point_with_search_range_index = {
-    stop_line_position, dst_search_range};
-  const double stop_line_margin = base_link2front + planner_param_.stop_margin;
   /**
    * @brief : calculate signed arc length consider stop margin from stop line
    *
    * |----------------------------|
    * s---ego----------x--|--------g
    */
+  const size_t stop_line_seg_idx = planning_utils::calcSegmentIndexFromPointIndex(
+    path->points, stop_pose.position, stop_point_idx);
+  const double stop_line_margin = base_link2front + planner_param_.stop_margin;
```

Comment:
I think `stop_line_margin` should not contain `base_link2front` because the offset has already considered in `stop_pose` calculation in `arc_lane_utils::createTargetPoint()`.

Reply:
`planner_param_.stop_margin` is also considered in that util function ?